src/js/util/walkthrough-utils.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+var _ = require('lodash');
+
+/**
+ * Walkthrough Utility has functions used in the walkthrough react components
+ * that need to be well tested and can exist outside of the ui
+ */
+
+/**
+ * The container for the property values in the store. These differ by primitive type.
+ * @TODO update these when guide and scales are added into redux
+ * @type {Object}
+ */
+var PROPERTY_CONTAINER = {
+  marks: 'properties.update',
+  guides: 'property',
+  scales: ''
+};
+
+/**
+ * Get the properties off the primitive
+ * @param  {Object} object - a primitive object
+ * @param  {String} type - either marks, guides or scales
+ * @returns {Object} - object of the primitives properties
+ */
+function getProperties(object, type) {
+  var props = PROPERTY_CONTAINER[type].split('.');
+  var propertyObj = object;
+  for (var i = 0; i < props.length; i++) {
+    propertyObj = propertyObj ? propertyObj[props[i]] : propertyObj;
+  }
+  return propertyObj;
+}
+
+/**
+ * Get an array of all elements with a certain type
+ * Wraps lodash filter
+ * @param  {string} elemType - a string that will match the key in each state element
+ * @param  {Array} stateArray - an array of objects to test against
+ * @param  {string} key - key to filter by
+ * @returns {[type]}            [description]
+ */
+function filterBy(elemType, stateArray, key) {
+  return _.filter(stateArray, function(o) {
```

Comment:
Does a regular `stateArray.filter` suffice here? Reduces our dependency on lodash, if we switch to their modular dependencies.